### PR TITLE
add loadbalancer unit test

### PR DESCRIPTION
**Pull request by DoubleLuXu**
## PR Type
Feature.

## Describe what this PR does for and how you did.

add loadbalancer unit test

## Adding the issue link (#xxx) if possible.

https://github.com/Tencent/spring-cloud-tencent/issues/421

## Checklist

- [ ] Add copyright holder at the beginning of .class file if it is new.
- [ ] Add information of this PR to CHANGELOG.md in root of project.
- [ ] All junit tests passing.
- [ ] Coverage from `Codecov Report` should not decrease.

## Checklist (Optional)

- [ ] Will Pull Request to branch of 2020.0 and 2021.0.
- [ ] Add documentation in javadoc in code or comment below the PR if necessary.
- [ ] Add your name as @author to the beginning of .class file.

### CODE REVIEW

**Review by lepdou — DISMISSED** [review]
LGTM

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import static com.tencent.cloud.common.metadata.MetadataContext.LOCAL_NAMESPACE;
+import static com.tencent.cloud.common.metadata.MetadataContext.LOCAL_SERVICE;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * Test for {@link PolarisLoadBalancer}
```
Comment (SkyeBeFreeman) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java:57]: 句尾添加句号

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import org.springframework.cloud.client.DefaultServiceInstance;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.loadbalancer.core.ServiceInstanceListSupplier;
+
+import static com.tencent.cloud.common.metadata.MetadataContext.LOCAL_NAMESPACE;
+import static org.mockito.ArgumentMatchers.anyString;
+
+/**
+ * Test for {@link PolarisServiceInstanceListSupplier}
```
Comment (SkyeBeFreeman) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java:42]: 句尾添加句号

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		verify(loadBalancerProperties).getEnabled();
+		verify(supplierObjectProvider).getIfAvailable(any());
+
+		//result assert
+		Assert.assertNotNull(serviceInstance);
+		Assert.assertTrue("polaris service instance",
+				serviceInstance instanceof PolarisServiceInstance);
+
+		PolarisServiceInstance polarisServiceInstance = (PolarisServiceInstance) serviceInstance;
+		Assert.assertEquals("instance-id", polarisServiceInstance.getPolarisInstance().getId());
```
Comment (SkyeBeFreeman) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java:122]: 使用`org.assertj.core.api.Assertions.assertThat`以和其他单测保持统一

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			allServers.add(instance2);
+
+			List<ServiceInstance> polarisInstanceList = instanceListSupplier.chooseInstances(allServers);
+			Assert.assertNotNull(polarisInstanceList);
+			Assert.assertEquals(polarisInstanceList.size(), allServers.size());
+			for (ServiceInstance serviceInstance : polarisInstanceList) {
+				Assert.assertTrue("", serviceInstance instanceof PolarisServiceInstance);
+				PolarisServiceInstance polarisServiceInstance = (PolarisServiceInstance) serviceInstance;
+
+				Assert.assertFalse(polarisServiceInstance.isSecure());
```
Comment (SkyeBeFreeman) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java:78]: 使用`org.assertj.core.api.Assertions.assertThat`以和其他单测保持统一

**Review by SkyeBeFreeman — CHANGES_REQUESTED** [review]

**Review by SkyeBeFreeman — DISMISSED** [review]
LGTM.

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+		PolarisServiceInstance polarisServiceInstance = (PolarisServiceInstance) serviceInstance;
+
+		Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getId()).isEqualTo("instance-id");
+		Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getNamespace()).isEqualTo(LOCAL_NAMESPACE);
+		Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getService()).isEqualTo(LOCAL_SERVICE);
+		Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getHost()).isEqualTo("host");
+		Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getPort()).isEqualTo(8090);
+	}
+
+}
```
Comment (misselvexu) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java:129]: Add a blank line at the end of the file .

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getService()).isEqualTo("unit-test-serviceId");
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getNamespace()).isEqualTo(LOCAL_NAMESPACE);
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getPort()).isEqualTo(8090);
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getId().startsWith("unit-test-instanceId")).isTrue();
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getHost().startsWith("unit-test-host")).isTrue();
+
+			}
+		}
+	}
+}
```
Comment (misselvexu) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java:91]: Add a blank line at the end of the file.

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * Test for {@link PolarisLoadBalancer}.
+ *
+ * @author rod.xu
+ * @date 2022/7/21 5:44 下午
```
Comment (misselvexu) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java:60]: Remove @date from the non-javadoc standard syntax.

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+import org.springframework.cloud.loadbalancer.core.ServiceInstanceListSupplier;
+
+import static com.tencent.cloud.common.metadata.MetadataContext.LOCAL_NAMESPACE;
+import static org.mockito.ArgumentMatchers.anyString;
+
+/**
+ * Test for {@link PolarisServiceInstanceListSupplier}.
+ *
+ * @author rod.xu
+ * @date 2022/7/21 5:45 下午
```
Comment (misselvexu) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java:45]: Remove @date from the non-javadoc standard syntax.

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+	@Mock
+	private RouterAPI routerAPI;
+	@Mock
+	private ObjectProvider<ServiceInstanceListSupplier> supplierObjectProvider;
+	@Mock
+	private PolarisLoadBalancerProperties loadBalancerProperties;
+
+	private static MockedStatic<ApplicationContextAwareUtils> mockedApplicationContextAwareUtils;
+	private static Instance TestInstance;
```
Comment (misselvexu) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisLoadBalancerTest.java:73]: Rename field `TestInstance` to `testInstance`.

File: spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+				PolarisServiceInstance polarisServiceInstance = (PolarisServiceInstance) serviceInstance;
+
+				Assertions.assertThat(polarisServiceInstance.isSecure()).isFalse();
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getService()).isEqualTo("unit-test-serviceId");
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getNamespace()).isEqualTo(LOCAL_NAMESPACE);
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getPort()).isEqualTo(8090);
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getId().startsWith("unit-test-instanceId")).isTrue();
+				Assertions.assertThat(polarisServiceInstance.getPolarisInstance().getHost().startsWith("unit-test-host")).isTrue();
+
```
Comment (misselvexu) [inline spring-cloud-tencent-polaris-loadbalancer/src/test/java/com/tencent/cloud/polaris/loadbalancer/PolarisServiceInstanceListSupplierTest.java:87]: Remove useless blank line .

**Review by misselvexu — CHANGES_REQUESTED** [review]

**Review by misselvexu — APPROVED** [review]

**Pull request merged**